src/target/stm32h7.c

```diff
@@ -264,7 +264,7 @@ bool stm32h7_probe(target_s *target)
 	/* By now it's established that this is likely an H7, but check that it's not an MP15x_CM4 with an errata in AP part code */
 	const uint32_t idcode = target_mem32_read32(target, STM32H7_DBGMCU_IDCODE);
 	const uint16_t dev_id = idcode & STM32H7_DBGMCU_IDCODE_DEV_MASK;
-	DEBUG_TARGET("%s: looking at device ID 0x%03x at 0x%08" PRIx32 "\n", __func__, dev_id, STM32H7_DBGMCU_IDCODE);
+	DEBUG_TARGET("%s: looking at device ID 0x%03" PRIx16 " at 0x%08x\n", __func__, dev_id, STM32H7_DBGMCU_IDCODE);
```

Comment:
`PRIx16` translates to "hx" for "%03hx" which doesn't work on newlib-nano - please retain the "x" ("%03x") specifier for uint16_t's here.

Does casting the STM32H7_DBGMCU_IDCODE constant to a uint32_t (with the original format string) fix the build error for you?

Reply:
Casting `STM32H7_DBGMCU_IDCODE` to a `uint32_t` does work. The `PRIx16` fix was due to me stumbling over the very verbose error.

I've updated the patch to have this change.